src/modules/mc_rate_control/MulticopterRateControl.cpp

```diff
@@ -173,12 +173,12 @@ MulticopterRateControl::Run()
 			if (_manual_control_setpoint_sub.update(&manual_control_setpoint)) {
 				// manual rates control - ACRO mode
 				const Vector3f man_rate_sp{
-					math::superexpo(manual_control_setpoint.y, _param_mc_acro_expo.get(), _param_mc_acro_supexpo.get()),
-					math::superexpo(-manual_control_setpoint.x, _param_mc_acro_expo.get(), _param_mc_acro_supexpo.get()),
-					math::superexpo(manual_control_setpoint.r, _param_mc_acro_expo_y.get(), _param_mc_acro_supexpoy.get())};
+					math::superexpo(manual_control_setpoint.roll, _param_mc_acro_expo.get(), _param_mc_acro_supexpo.get()),
+					math::superexpo(-manual_control_setpoint.pitch, _param_mc_acro_expo.get(), _param_mc_acro_supexpo.get()),
+					math::superexpo(manual_control_setpoint.yaw, _param_mc_acro_expo_y.get(), _param_mc_acro_supexpoy.get())};
 
 				_rates_setpoint = man_rate_sp.emult(_acro_rate_max);
-				_thrust_setpoint(2) = -math::constrain(manual_control_setpoint.z, 0.0f, 1.0f);
+				_thrust_setpoint(2) = -(manual_control_setpoint.throttle + 1.f) * .5f;
```

Comment:
Maybe a dumb question, but how do we handle reverse thrusts? Do we even support them in PX4? Cuz this logic won't give negative thrust setpoint :thinking:, and I am wondering how that would result in reversable ESCs.

Reply:
Discussed: Reverse thrust commands for quads aren't supported now. In the future, we can add it like:

```c++
if (reverse_thrust_setting) {
// allow reverse thrust
} else {
// current logic, only allow positive thrust
}
```